agent/structs/config_entry_discoverychain.go

```diff
@@ -807,6 +829,63 @@ func (e *ServiceResolverConfigEntry) Validate() error {
 		return fmt.Errorf("Bad ConnectTimeout '%s', must be >= 0", e.ConnectTimeout)
 	}
 
+	if e.LoadBalancer != nil && e.LoadBalancer.EnvoyLBConfig != nil {
+		ec := e.LoadBalancer.EnvoyLBConfig
+
+		validPolicies := map[string]bool{
```

Comment:
nit: this could be a package scoped variable rather than stack allocating every call